xcode/Subconscious/Shared/Components/Detail/DetailMetaSheet.swift

```diff
@@ -0,0 +1,328 @@
+    case requestUpdateAudience(_ audience: Audience)
+    case succeedUpdateAudience(_ receipt: MoveReceipt)
+    
+    //  Delete entry requests
+    /// Show/hide delete confirmation dialog
+    case presentDeleteConfirmationDialog(Bool)
+    /// Request this address be deleted.
+    /// Should be handled by parent component.
+    case requestDelete(MemoAddress?)
+
+    static var refreshRenameSuggestions: Self {
+        .renameSearch(.refreshRenameSuggestions)
+    }
+    
+    static func setRenameSearchSubject(_ address: MemoAddress?) -> Self {
+        .renameSearch(.setSubject(address))
+    }
+}
+
+struct DetailMetaSheetModel: ModelProtocol {
+    typealias Action = DetailMetaSheetAction
+    typealias Environment = AppEnvironment
+    
+    var address: MemoAddress?
+    var defaultAudience = Audience.local
+    var audience: Audience {
+        address?.toAudience() ?? defaultAudience
+    }
+    var isRenameSheetPresented = false
+    var renameSearch = RenameSearchModel()
+    
+    /// Is delete confirmation dialog presented?
+    var isDeleteConfirmationDialogPresented = false
+    
+    static func update(
+        state: Self,
+        action: Action,
+        environment: Environment
+    ) -> ObservableStore.Update<Self> {
+        switch action {
+        case .renameSearch(let action):
+            return DetailMetaSheetRenameSearchCursor.update(
+                state: state,
+                action: action,
+                environment: environment
+            )
+        case .presentRenameSheet(let isPresented):
+            return presentRenameSheet(
+                state: state,
+                environment: environment,
+                isPresented: isPresented
+            )
+        case .presentRenameSheetFor(let address):
+            return update(
+                state: state,
+                actions: [
+                    .setRenameSearchSubject(address),
+                    .presentRenameSheet(true)
+                ],
+                environment: environment
+            )
+        case .selectRenameSuggestion(let suggestion):
+            return selectRenameSuggestion(
+                state: state,
+                environment: environment,
+                suggestion: suggestion
+            )
+        case .presentDeleteConfirmationDialog(let isPresented):
+            return presentDeleteConfirmationDialog(
+                state: state,
+                environment: environment,
+                isPresented: isPresented
+            )
+        case let .setAddress(address):
+            return setAddress(
+                state: state,
+                environment: environment,
+                address: address
+            )
+        case .setDefaultAudience(let audience):
+            return setDefaultAudience(
+                state: state,
+                environment: environment,
+                audience: audience
+            )
+        case .requestUpdateAudience:
```

Comment:
Might be worth a comment explaining where this is actually handled?

Reply:
Noted in action definition. https://github.com/subconsciousnetwork/subconscious/pull/427/files/55bcffaa2f77af4ec15e64cdd1ec5989c006d163#diff-cc3a094b36c3ad6fcf4fd7f720a74198f020e94eb390d0cd45d19b14456a64afR125

I prefer to have these comments at the point of action definition, because they often bubble up in Xcode's code helpers when written there.